platform/security/core/security-core-services/src/main/java/ddf/security/permission/impl/PermissionsImpl.java

```diff
@@ -72,4 +78,27 @@ public PermissionsImpl() {}
     }
     return stringAttributes;
   }
+
+  @Override
+  public KeyValueCollectionPermission buildKeyValueCollectionPermission(
+      String action, KeyValuePermission... permissions) {
+    return new KeyValueCollectionPermissionImpl(action, permissions);
+  }
+
+  @Override
+  public KeyValueCollectionPermission buildKeyValueCollectionPermission(
+      String action, Map<String, ? extends Collection<String>> map) {
+    return new KeyValueCollectionPermissionImpl(action, map);
+  }
+
+  @Override
+  public KeyValueCollectionPermission buildKeyValueCollectionPermission(
+      String action, Collection<KeyValuePermission> permissions) {
+    return new KeyValueCollectionPermissionImpl(action, permissions);
+  }
+
+  @Override
+  public KeyValuePermission buildKeyValuePermission(String key, Set<String> values) {
+    return new KeyValuePermissionImpl(key, values);
+  }
```

Comment:
✏️ These builder methods **might** be better as static methods on the interface.

Pro: when you want to call one of these builders, you don't have to inject a `Permissions` service into the java class. You just call the static interface method.
Con: testing

Reply:
Can't do a static method without embedding the bundle or directly referencing an impl of some sort via an import.